z/btree.go

```diff
@@ -65,30 +73,41 @@ func (t *Tree) initRootNode() {
 }
 
 // NewTree returns a memory mapped B+ tree with given filename.
-func NewTree(fname string, maxSz int) *Tree {
+func NewTree(fname string, maxSz, mlockSz int) *Tree {
 	mf, err := createFile(maxSz, fname)
 	if err != NewFile {
 		check(err)
 	}
-	// Tell kernel that we'd be reading pages in random order, so don't do read ahead.
-	check(Madvise(mf.Data, false))
-
 	t := &Tree{
 		mf:       mf,
 		nextPage: 1,
+		mlockSz:  mlockSz,
 	}
+	check(Madvise(t.mf.Data, false))
+	check(Mlock(t.mf.Data, t.mlockSz))
 	t.initRootNode()
 	return t
 }
 
+func (t *Tree) truncate(toSz int64) {
+	check(Munlock(t.mf.Data, t.mlockSz))
```

Comment:
Munlock() is not required and can be removed. It is implicitly called on truncate. Though, I couldn't find any doc but verified locally.